web3/utils/encoding.py

```diff
@@ -1,29 +1,117 @@
 # String encodings and numeric representations
+import sys
 import json
+import codecs
 
 from rlp.sedes import big_endian_int
 
 from eth_utils import (
-    coerce_args_to_bytes,
-    coerce_args_to_text,
-    encode_hex,
-    is_0x_prefixed,
+    is_bytes,
+    is_string,
     is_boolean,
     is_dict,
     is_integer,
-    is_string,
-    is_bytes,
+    coerce_args_to_text,
+    coerce_args_to_bytes,
+    add_0x_prefix,
+    is_0x_prefixed,
+    encode_hex,
+    remove_0x_prefix,
     force_text,
     force_bytes,
 )
 
+from web3.utils.abi import (
+    is_address_type,
+    is_array_type,
+    is_bool_type,
+    is_bytes_type,
+    is_int_type,
+    is_uint_type,
+    is_string_type,
+    size_of_type,
+    sub_type_of_array_type,
+)
+
+from web3.utils.validation import (
+    validate_abi_type,
+    validate_abi_value,
+)
+
 
 def _is_prefixed(value, prefix):
     return value.startswith(
         force_bytes(prefix) if is_bytes(value) else force_text(prefix)
     )
 
 
+def hex_encode_abi_type(abi_type, value, force_size=None):
+    """
+    Encodes value into a hex string in format of abi_type
+    """
+    validate_abi_type(abi_type)
+    validate_abi_value(abi_type, value)
+
+    data_size = force_size or size_of_type(abi_type)
+    if is_array_type(abi_type):
+        sub_type = sub_type_of_array_type(abi_type)
+        return "".join([remove_0x_prefix(hex_encode_abi_type(sub_type, v, 256)) for v in value])
+    elif is_bool_type(abi_type):
+        return to_hex_with_size(value, data_size)
+    elif is_uint_type(abi_type):
+        return to_hex_with_size(value, data_size)
+    elif is_int_type(abi_type):
+        return to_hex_twos_compliment(value, data_size)
+    elif is_address_type(abi_type):
+        return pad_hex(value, data_size)
+    elif is_bytes_type(abi_type):
+        if sys.version_info[0] >= 3:
+            if is_bytes(value):
+                return to_hex(value)
+            return value
+        else:
+            return value
+    elif is_string_type(abi_type):
+        return bytes_to_hex(value)
+
+
+def bytes_to_hex(byte_string):
+    """
+    Converts byte_string to hex representation
+    """
+    if type(byte_string) == str:
```

Comment:
I think that the `eth_utils.is_text` function does the same thing.  If not, then can you convert this to `isinstance(byte_string, str)`.